### DISCUSSION TITLE
News sidebar shows months in the wrong year

### DISCUSSION THREAD

**Discussion by Gabixel** [ppy/osu#16504] [Q&A] [answered]
The news sidebar shows November and December 2022. The posts from those months seem to be from 2021 (from the same months):
![image](https://user-images.githubusercontent.com/43073074/149936434-9bd5534d-da30-483b-b02c-dce64bb505fc.png)

Filtering in 2021 confirms this:
![image](https://user-images.githubusercontent.com/43073074/149936455-450b737a-4323-492b-9892-fb29f7ae910e.png)

For comparison, from web it looks like this:
![image](https://user-images.githubusercontent.com/43073074/149937348-cc2d74d6-d259-4b8f-a31e-7feedbd90764.png)

**Comment by frenzibyte** [comment]
Something sounds off here.. why would the "2022" filter in web api show posts published at 2021 in the first place?

**Reply by bdach** [reply]
the initial request is not scoped to any year at all, it's just a rolling feed.

https://github.com/ppy/osu/blob/fffa3550976c49a3e6c8b0f3ad5c1289bc39879d/osu.Game/Overlays/NewsOverlay.cs#L80

https://github.com/ppy/osu/blob/1c560a949051d6103db9ecd87892141e1f9de5f0/osu.Game/Online/API/Requests/GetNewsRequest.cs#L25-L26

the client implementation of the view is mistaken in its assumptions.

**Reply by frenzibyte** [reply]
Though using the API directly and adding `year=2022`, I still get posts from the last two months of the previous year. Feels like weird UX if it's intentional.

**Reply by bdach** [reply]
that one is a bit weirder, i agree.

**Comment by bdach** [comment, marked as answer]
Addressed by #16526
